Add Arch Linux official WSL image to the distribution manifest

Antiz96 opened the pull request:
As discussed in https://github.com/microsoft/WSL/issues/12551, we (Arch Linux) now [provide an official WSL image](https://rfc.archlinux.page/0050-arch-linux-wsl-image/). This commit adds it to the Microsoft WSL's official distribution manifest, allowing users to pull and install it in an automated way (`wsl --install archlinux`).

Closes https://github.com/microsoft/WSL/issues/12551

Antiz96 commented:
Our image is compressed with `zstd` (.tar.zst) which the `distributions/validate-modern.py` script doesn't seem to support currently.  
If that's of interest, I can look at adding support for it to the script :)

OneBlue commented:
Thank you for doing this @Antiz96. 

Unfortunately WSL indeed doesn't officially support zstd compression. The version of bsdtar we use in WSL2 happens to support it, but WSL1 doesn't. For instance, if I try to install Archlinux in WSL1 mode I get: 

```
λ wsl --install --from-file  archlinux-2025.04.01.123847.wsl --version 1
Installing: archlinux-2025.04.01.123847.wsl
bsdtar: Error opening archive: Unrecognized archive format ]
Importing the distribution failed.
Error code: Wsl/Service/RegisterDistro/WSL_E_IMPORT_FAILED
```

We usually recommend distributions to compress with gz, or xz. Would that be an option for Archlinux ?

Antiz96 commented:
> Unfortunately WSL indeed doesn't officially support zstd compression.

Oh, I wasn't aware of this. Sorry :hand_over_mouth: 

> We usually recommend distributions to compress with gz, or xz. Would that be an option for Archlinux ?

Yes, that should be doable :)
I will adapt our build system accordingly and update this PR with an image compressed either with gz or xz shortly.

Antiz96 committed:
Add Arch Linux official WSL image to the distribution manifest

As discussed in microsoft#12551, we (Arch Linux) now [provide an official WSL image](https://rfc.archlinux.page/0050-arch-linux-wsl-image/). This commit adds it to the Microsoft WSL's official distribution manifest, allowing users to pull and install it in an automated way (`wsl --install archlinux`).

Closes microsoft#12551

OneBlue reviewed (approved):
LGTM !

craigloewen-msft commented:
LGTM too! :) Thanks for the awesome contribution!!
